### DISCUSSION TITLE
About Initialization of Android in the version 11.0.0-preview3

### DISCUSSION THREAD

**Discussion by CYLMos** [AvaloniaUI/Avalonia#9321] [Q&A] [answered]
Today I updated the package from 0.10.18 to 11.0.0-preview3 and I got the issue.

I noticed I should change AvaloniaActivity<App> to AvaloniaMainActivity in MainActivity.cs and Activity to AvaloniaSplashActivity<App> in SplashActivity.cs. But I can't understand the initialization procedure.

And the issue I have is that I got black screen after switched from the splash activity

In my App.axaml.cs, it would judge the current platform and initialize.
```
public sealed partial class App : Application
    {
        public override void Initialize()
        {
            AvaloniaXamlLoader.Load(this);
        }

        public override void OnFrameworkInitializationCompleted()
        {
            if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
            {
                desktop.MainWindow = new MainWindow();
            }
            else if (ApplicationLifetime is ISingleViewApplicationLifetime mobile)
                mobile.MainView = new MainView();

            base.OnFrameworkInitializationCompleted();
        }
    }
```

And in SpalshActivity.cs, `CustomizeAppBuilder(Avalonia.AppBuilder builder)` is moved to AvaloniaSplashActivity
```
[Activity(Theme = "@style/MyTheme.Splash", MainLauncher = true, NoHistory = true)]
public sealed class SplashActivity : AvaloniaSplashActivity<App>
{
        protected override Avalonia.AppBuilder CustomizeAppBuilder(Avalonia.AppBuilder builder)
        {
            return base.CustomizeAppBuilder(builder);
        }

        protected override void OnResume()
        {
            base.OnResume();

            StartActivity(new Intent(Application.Context, typeof(MainActivity)));
        }
}
```
So `OnFrameworkInitializationCompleted()` in App.axaml.cs would be executed when SplashActivity shows up.
And then SplashActivity would be disappeared and MainActivity shows up.
````
[Activity(Label = "HelloWorld.Android", Theme = "@style/MyTheme.NoActionBar", Icon = "@drawable/icon", LaunchMode = LaunchMode.SingleTop, ConfigurationChanges = ConfigChanges.Orientation | ConfigChanges.ScreenSize)]
    public sealed class MainActivity : AvaloniaMainActivity
    {
        protected override void OnResume()
        {
            base.OnResume();
            // Do somethings 
        }
    }
````
I also tried to add break points in `OnResum()` to make sure all the codes in this function are good.
After that, the screen is always black.

**Comment by CYLMos** [comment, marked as answer]
Ok, I'm dummy. The problem is in MainView and some logic error happen there.

**Reply by timunie** [reply]
Thank you for sharing your findings with the community 👍
